### ISSUE TITLE
ENV after SHELL resets Hadolint's tracked shell and causes false SC3010 warnings

### ISSUE THREAD

**Issue by sebthom** [hadolint/hadolint#1241]
### Hadolint version

2.15.0

This appears to be a regression from 2.14.0.

### Minimal reproduction

```Dockerfile
FROM debian:trixie-slim

SHELL ["/bin/bash", "-euo", "pipefail", "-c"]

ENV FOO=bar

RUN <<EOF
  if [[ -n "$FOO" ]]; then
    echo "$FOO"
  fi
EOF
```

Run: `docker run --rm -i ghcr.io/hadolint/hadolint:v2.15.0-debian < Dockerfile`

Actual result: `Dockerfile:7 SC3010 warning: In POSIX sh, [[ ]] is undefined.`

Expected result: `No SC3010 warning. Docker's SHELL instruction affects subsequent shell-form RUN instructions, including heredocs. ENV does not reset the selected shell, so this script runs with Bash.`

### Additional findings

Moving the SHELL instruction below ENV, or repeating it after ENV, removes the warning.

The likely cause is addVars in Hadolint.Rule.Shellcheck. When processing an ENV or ARG, it constructs new shell options using:
```hs
shellName = Shell.shellName Shell.defaultShellOpts
```
This resets Hadolint's tracked shell to /bin/sh -c. It appears that this should preserve Shell.shellName opts instead.

Source:

https://github.com/hadolint/hadolint/blob/v2.15.0/src/Hadolint/Rule/Shellcheck.hs#L110-L123

**Comment by m-ildefons** [comment]
Hi @sebthom ,

thanks for reporting this. Seems like this is the same problem as https://github.com/hadolint/hadolint/issues/1228
That defect has already been fixed on `master` branch: https://github.com/hadolint/hadolint/blob/3e9d30d87cc977bbca7448a7b80094bca9692f06/src/Hadolint/Rule/Shellcheck.hs#L120
